src/dolphin/downloadDolphin.ts

```diff
@@ -27,106 +39,63 @@ export async function assertDolphinInstallation(
+    const data = await fetchLatestDolphin(type);
+    await downloadAndInstallDolphin(type, data, log);
   }
 }
 
-export async function downloadAndInstallDolphin(
-  type: DolphinLaunchType,
-  log: (message: string) => void,
-  cleanInstall = false,
-): Promise<void> {
-  const downloadedAsset = await downloadLatestDolphin(type, log);
-  log(`Installing ${type} Dolphin...`);
-  await installDolphin(type, downloadedAsset, log, cleanInstall);
-  log(`Finished ${type} installing`);
-}
-
-export async function assertDolphinInstallations(): Promise<void> {
-  try {
-    await assertDolphinInstallation(DolphinLaunchType.NETPLAY, logDownloadInfo);
-    await assertDolphinInstallation(DolphinLaunchType.PLAYBACK, logDownloadInfo);
-    await ipc_dolphinDownloadFinishedEvent.main!.trigger({ error: null });
-  } catch (err) {
-    console.error(err);
-    await ipc_dolphinDownloadFinishedEvent.main!.trigger({ error: err.message });
-  }
-  return;
-}
-
 async function compareDolphinVersion(type: DolphinLaunchType, latestVersion: string): Promise<boolean> {
   const dolphinPath = await findDolphinExecutable(type);
   const dolphinVersion = spawnSync(dolphinPath, ["--version"]).stdout.toString();
   return lt(dolphinVersion, latestVersion);
 }
 
-export async function openDolphin(type: DolphinLaunchType, params?: string[]): Promise<ChildProcessWithoutNullStreams> {
-  const dolphinPath = await findDolphinExecutable(type);
-  return spawn(dolphinPath, params);
-}
-
-async function getLatestDolphinAsset(type: DolphinLaunchType): Promise<any> {
-  const release = await getLatestReleaseData(type);
-  const asset = release.assets.find((a: any) => matchesPlatform(a.name));
-  if (!asset) {
-    throw new Error(`No release asset matched the current platform: ${process.platform}`);
-  }
-  return asset;
-}
-
-async function getLatestReleaseData(type: DolphinLaunchType): Promise<any> {
-  const owner = "project-slippi";
-  let repo = "Ishiiruka";
-  if (type === DolphinLaunchType.PLAYBACK) {
-    repo += "-Playback";
-  }
-  return getLatestRelease(owner, repo);
-}
-
-function matchesPlatform(releaseName: string): boolean {
-  switch (process.platform) {
-    case "win32":
-      return releaseName.endsWith("Win.zip");
-    case "darwin":
-      return releaseName.endsWith("Mac.dmg");
-    case "linux":
-      return releaseName.endsWith("Linux.zip");
-    default:
-      return false;
-  }
+export async function downloadAndInstallDolphin(
+  type: DolphinLaunchType,
+  releaseInfo: DolphinVersionResponse,
+  log: (message: string) => void,
+  cleanInstall = false,
+): Promise<void> {
+  const downloadedAsset = await downloadLatestDolphin(releaseInfo, log);
+  log(`Installing v${releaseInfo.version} ${type} Dolphin...`);
+  await installDolphin(type, downloadedAsset, log, cleanInstall);
+  log(`Finished v${releaseInfo.version} ${type} Dolphin install`);
 }
 
 async function downloadLatestDolphin(
-  type: DolphinLaunchType,
+  releaseInfo: DolphinVersionResponse,
```

Comment:
```suggestion
  downloadUrl: string,
```
Can we pass the URL in directly here?